lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -2716,6 +2716,37 @@ static llvm::Expected<addr_t> ReadAsyncContextRegisterFromUnwind(
       "SwiftLanguageRuntime: Unsupported register location type = %d", loctype);
 }
 
+/// Returns true if the async register should be dereferenced once to obtain the
+/// CFA of the currently executing function. This is the case at the start of
+/// "Q" funclets, before the low level code changes the meaning of the async
+/// register to not require the indirection.
+/// This implementation detects the transition point by comparing the
+/// continuation pointer in the async context with the currently executing
+/// funclet given by the SymbolContext sc. If they are the same, the PC is
+/// before the transition point.
+/// FIXME: this fails in some recursive async functions. See: rdar://139676623
+static llvm::Expected<bool> IsIndirectContext(Process &process,
+                                              StringRef mangled_name,
+                                              addr_t async_reg,
```

Comment:
Just out of curiosity: what is this? Is it a DWARF register number stored in an `addr_t`?

Reply:
it is the contents of the async reg. I hesitated to call it "async context" because we don't know whose context this is (this is what the function is indirectly determining)